### DISCUSSION TITLE
sleep function?

### DISCUSSION THREAD

**Discussion by NjuuOwO** [YimMenu/YimMenu#1984] [Q&A] [answered]
Is there like Sleep(int ms) function within the menu?

**Comment by ghost** [comment]
`std::this_thread::sleep_for(100ms);` No?
If you are talking about lua, then an example of use:
`
script.register_looped("nameOfMyLoopedScript", function (script)
     -- sleep until next game frame
     script:yield()

     local ModelHash = joaat("adder")
     if not STREAMING.IS_MODEL_IN_CDIMAGE(ModelHash) then return end
     STREAMING.REQUEST_MODEL(ModelHash) -- Request the model
     while not STREAMING.HAS_MODEL_LOADED(ModelHash) do -- Waits for the model to load
         script:yield()
     end
     local myPed = PLAYER.PLAYER_PED_ID()
     local myCoords = ENTITY.GET_ENTITY_COORDS(myPed, true)
     -- Spawns a networked vehicle on your current coords
     local spawnedVehicle = VEHICLE.CREATE_VEHICLE(ModelHash, myCoords.x, myCoords.y, myCoords.z, ENTITY.GET_ENTITY_HEADING(myPed), true, false)
     -- removes model from game memory as we no longer need it
     STREAMING.SET_MODEL_AS_NO_LONGER_NEEDED(ModelHash)
     -- sleep for 2s
     script:sleep(2000)
     ENTITY.DELETE_ENTITY(spawnedVehicle)
end)
`

**Reply by NjuuOwO** [reply]
I'm talking about C++

> `std::this_thread::sleep_for(100ms);`

I was trying to use that, but it made entire game process freeze for set period of time. Maybe i did run it wrong inside player_command, idrk, so I tried running it inside `std::thread()`, but it had another problem of the thread exiting when another action was made (e.g. enable invisibility when sleep_for() runs)

Right now I'm using
`for (int i = 0; i < 30; ++i)
script::get_current()->yield();`
but this makes the delay not static as it can range from 1000ms to 200ms (depending on games framerate NOWAY) which so I'm looking for a time based sleep.

**Reply by xiaoxiao921** [reply, marked as answer]
`std::this_thread::sleep_for(100ms);` inside a game script thread will make the game hang yes, obviously, if you want to just sleep (well, yield until a given amount of time) you can just use https://github.com/YimMenu/YimMenu/blob/a47af68384040e6330b863dcf837026eaf4afcf4/src/backend/looped/weapons/steal_vehicle_gun.cpp#L27

`script::get_current()->yield(5s);  // for 5s ` also work, std chrono is available on cppreference for doc

there is literally ton of usage of yield with std chrono parameter around, use string search inside the code base

**Reply by NjuuOwO** [reply]
Aah I see. Thanks for help ;d
